### PR TITLE
Notification Plugin docs

### PR DESCRIPTION
**Pull request by dangeross**
This PR adds docs for registering a webhook and integrating the notification plugin

Closes https://github.com/breez/breez-sdk-liquid/issues/444

### CODE REVIEW

File: src/notifications/android_plugin.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        return readSecuredValue(applicationContext, ACCOUNT_MNEMONIC)
+            ?.let { mnemonic ->
+                ConnectRequest(config, mnemonic)
+            }
+    }
+}
+```
+
+## Reference implementation
+For a complete reference, see how we implemented it in misty-breez wallet: [BreezFcmService.kt](https://github.com/breez/misty-breez/blob/main/android/app/src/main/kotlin/com/cBreez/client/BreezFcmService.kt).
```
Comment (ok300) [inline src/notifications/android_plugin.md:128]: This gives a 404 error.

File: src/notifications/ios_plugin.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            os_log(.error, "Mnemonic not found")
+            return nil
+        }
+        return ConnectRequest(config: config, mnemonic: mnemonic)
+    }
+}
+```
+
+## Reference implementation
+For a complete reference, see how we implemented it in misty-breez wallet: [NotificationService.swift](https://github.com/breez/misty-breez/blob/main/ios/Breez%20Notification%20Service%20Extension/NotificationService.swift).
```
Comment (ok300) [inline src/notifications/ios_plugin.md:70]: Also gives a 404 error.

**Review by ok300 — COMMENTED** [review]
Looks good, just a few stale links and one NIT:

It would be clearer to use `id_hash` instead of `id` in
```json
    "data": {  
        "id": "",    // The hashed id of the swap
        "status": "" // The latest swap status
    }
```

as it could avoid confusion down the road and it's consistent with the SDK Greenlight notifications:

```json
    "data": {  
        "payment_hash": "" // The payment hash that is in progress
    }
```

If it's not too much effort, maybe it's worth changing it.

**Comment by dangeross** [comment]
> It would be clearer to use `id_hash` instead of `id` in

It's the data as sent from boltz via the webhook

File: src/notifications/android_plugin.md (hunk 1 above)
Comment (dangeross) [reply src/notifications/android_plugin.md:128]: These reference implementations I'm working on adding them now, so they're be not available for a bit. But I do see an error in this link, will correct it

File: src/SUMMARY.md [hunk 3]
```diff
@@ -22,3 +22,27 @@
...
+      - [Adding the Notification Plugin](notifications/ios_plugin.md)
+
+---
+
+# Advanced 
+
+- [Adding logging](notifications/logging.md)
+- [Changing default strings](notifications/changing_strings.md)
+- [Customising push messages](notifications/custom_messages.md)
+- [Handling custom notifications](notifications/custom_notifications.md)
```
Comment (erdemyerebasmaz) [inline src/SUMMARY.md:48]: ⚠️ Needs revision from other parties.

There's been one thing bothering me about titles, differences in cases. 
My suggestion is to be consistent and use Title Case on all these sections.
```suggestion
- [Implementing Mobile Notifications](notifications/getting_started.md)
- [Setting Up an NDS](notifications/setup_nds.md)
- [Using Webhooks](notifications/using_webhooks.md)
- [Integrating the Plugin](notifications/setup_plugin.md)
    - [Android](notifications/android_setup.md)
      - [Setting Up the Foreground Service](notifications/android_service.md)
      - [Adding the Notification Plugin](notifications/android_plugin.md)
    - [iOS](notifications/ios_setup.md)
      - [Setting Up the Notification Service Extension](notifications/ios_service.md)
      - [Adding the Notification Plugin](notifications/ios_plugin.md)

---

# Advanced 

- [Adding Logging](notifications/logging.md)
- [Changing Default Strings](notifications/changing_strings.md)
- [Customising Push Messages](notifications/custom_messages.md)
- [Handling Custom Notifications](notifications/custom_notifications.md)
```

File: src/guide/connecting.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 The first step is to construct the SDK configuration. Among others, it sets the network you want to use and the SDK working directory.
 
-The SDK uses the config `working_dir` to store the state of the SDK instance. When handling multiple instances of the SDK, one per node, each needs to have a different working directory defined.
+The SDK uses the config `working_dir` to store the state of the SDK instance. When handling multiple instances of the SDK each needs to have a different working directory defined.
```
Comment (erdemyerebasmaz) [inline src/guide/connecting.md:5]:
```suggestion
The SDK uses the config `working_dir` to store the state of the SDK instance. When handling multiple instances of the SDK, each instance needs to have a different working directory defined.
```

File: src/guide/lnurl_pay_service.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+}
+```
+
+The <b>reply_url</b> is used by the app to respond to the lnurlpay request.
+The <b>callback_url</b> s the LNURL-Pay callback URL, used by the payer to fetch the invoice.
+
+### Step 3: Responding to the callback url
+When the app receives the push notification, it parses the payload and then uses the reply_url to respond with the required data, for example:
+
+```
```
Comment (erdemyerebasmaz) [inline src/guide/lnurl_pay_service.md:65]:
```suggestion
```json
```

File: src/guide/lnurl_pay_service.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+### Step 2: Processing an LNURL-Pay request
+When an LNURL-Pay request is triggered a GET request to:
+```
+https://app.domain.com/lnurlp/[pubkey]
+```
+The service then sends a push notification to the app with the LNURL-Pay request and a callback URL. Such payload may look like the following:
+
+```
```
Comment (erdemyerebasmaz) [inline src/guide/lnurl_pay_service.md:49]:
```suggestion
```json
```

File: src/guide/lnurl_pay_service.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+### Step 1: Registering for an LNURL-Pay service
+Use a POST request to the service endpoint:
+
+```
+https://app.domain/lnurlpay/[pubkey]
+```
+With the following payload:
+
+```
```
Comment (erdemyerebasmaz) [inline src/guide/lnurl_pay_service.md:24]: We can add syntax highlighting for these json code blocks.
```suggestion
```json
```

File: src/guide/lnurl_pay_service.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+ "webhook_url": "notification service webhook url",
+ "signature": "signed payload"
+}
+```
+
+to register the app for an LNURL-Pay service.
+The signature is refers to the following string: ```[time]-[webhook_url]``` where ```time``` and ```webhook_url``` are the payload fields.
+
+The service responds with following payload: 
+```>
```
Comment (erdemyerebasmaz) [inline src/guide/lnurl_pay_service.md:36]: **Note:** I don't know what `>` stands for here.
```suggestion
```json
```

File: src/guide/lnurl_pay_service.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
+# Receiving payments using LNURL-Pay
+
+Breez SDK - *Liquid* users have the ability to receive Lightning payments using [LNURL-Pay](https://github.com/lnurl/luds/blob/luds/06.md).
+
+LNURL-Pay requires a web service that serves LNURL-Pay requests. This service needs to communicate with the SDK in order to fetch the necessary metadata data and the associated payment request.
+To interact with the SDK, the service uses a simple protocol over push notifications:
+* The service sends a push notification to the user's mobile app with the LNURL-Pay request and a reply URL.
+* The app responds to reply URL with the required data.
+* The data is forwarded from the service to the payer.
```
Comment (erdemyerebasmaz) [inline src/guide/lnurl_pay_service.md:9]: To be more inline with other items; 
- **service** _sends_
- **app** _responds_
```suggestion
* The service forwards the data to the payer.
```

File: src/guide/lnurl_pay_service.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+
+## General workflow
+The following workflow is application specific and the steps detailed below refer to the c-breez wallet implementation which requires running <b>[breez-lnurl](https://github.com/breez/breez-lnurl) </b>service.
```
Comment (erdemyerebasmaz) [inline src/guide/lnurl_pay_service.md:12]: Misty Breez?
```suggestion
The following workflow is application specific and the steps detailed below refer to the misty-breez wallet implementation which requires running <b>[breez-lnurl](https://github.com/breez/breez-lnurl) </b>service.
```

File: src/guide/lnurl_pay_service.md [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+```
+{
+ "time": "seconds since epoch",
+ "webhook_url": "notification service webhook url",
+ "signature": "signed payload"
+}
+```
+
+to register the app for an LNURL-Pay service.
+The signature is refers to the following string: ```[time]-[webhook_url]``` where ```time``` and ```webhook_url``` are the payload fields.
```
Comment (erdemyerebasmaz) [inline src/guide/lnurl_pay_service.md:33]:
```suggestion
The signature it refers to the following string: ```[time]-[webhook_url]``` where ```time``` and ```webhook_url``` are the payload fields.
```

File: src/guide/lnurl_pay_service.md [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+ "lnurl": "https://app.domain.com/lnurlp/[pubkey]", 
+}
+```
+
+### Step 2: Processing an LNURL-Pay request
+When an LNURL-Pay request is triggered a GET request to:
+```
+https://app.domain.com/lnurlp/[pubkey]
+```
+The service then sends a push notification to the app with the LNURL-Pay request and a callback URL. Such payload may look like the following:
```
Comment (erdemyerebasmaz) [inline src/guide/lnurl_pay_service.md:47]:
```suggestion
The service then sends a push notification to the app with the LNURL-Pay request and a callback URL. The payload may look like the following:
```

File: src/guide/lnurl_pay_service.md [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+  "callback_url": https://app.domain.com/lnurlpay/invoice
+  }
+}
+```
+
+The <b>reply_url</b> is used by the app to respond to the lnurlpay request.
+The <b>callback_url</b> s the LNURL-Pay callback URL, used by the payer to fetch the invoice.
+
+### Step 3: Responding to the callback url
+When the app receives the push notification, it parses the payload and then uses the reply_url to respond with the required data, for example:
```
Comment (erdemyerebasmaz) [inline src/guide/lnurl_pay_service.md:63]:
```suggestion
When the app receives the push notification, it parses the payload and then uses the <b>reply_url</b> to respond with the required data, for example:
```

File: src/guide/lnurl_pay_service.md [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+
+The service receives the response from the app and forwards it to the sender.
+
+### Step 4: Fetching a bolt11 invoice
+
+The sender fetches a bolt11 invoice by invoking a GET request to the callback_url with adding a specific amount as a query parameter. For example: 
+```
+https://app.domain.com/lnurlpay/invoice?amount=1000
+```
+An additional push notification is triggered to send the invoice request to the app. Then the app replied with the bolt11 invoice data.
```
Comment (erdemyerebasmaz) [inline src/guide/lnurl_pay_service.md:83]: Either option is fine, replies-responds:
```suggestion
An additional push notification is triggered to send the invoice request to the app. Then the app replies with the bolt11 invoice data.
```
```suggestion
An additional push notification is triggered to send the invoice request to the app. Then the app responds with the bolt11 invoice data.
```

File: src/notifications/android_setup.md [hunk 15]
```diff
@@ -0,0 +1,7 @@
+# Android
+
+In order to add the Notification Plugin to your Android application, first you need to setup the Foreground Service, then add the Notification Plugin dependency and integrate it:
+
+## Next steps
+- **[Setting up the Foreground Service](android_service.md)**
```
Comment (erdemyerebasmaz) [inline src/notifications/android_setup.md:6]: ⚠️ Exemplary suggestion.

If previous suggestion to use title cases on titles is to be implemented, it should affect all other links and the titles on the pages themselves.
```suggestion
- **[Setting Up the Foreground Service](android_service.md)**
```

File: src/notifications/changing_strings.md [hunk 16]
```diff
@@ -0,0 +1,31 @@
+# Changing default strings
```
Comment (erdemyerebasmaz) [inline src/notifications/changing_strings.md:1]: ⚠️ Exemplary suggestion.

If previous suggestion to use title cases on titles is to be implemented, it should affect all other links and the titles on the pages themselves.
```suggestion
# Changing Default Strings
```

File: src/notifications/custom_notifications.md [hunk 17]
```diff
@@ -0,0 +1,233 @@
+# Handling custom notifications
+
+You can customise the Notification Plugin even further by handling your own notification types sent to the application via push notification. Follow the code example below for adding these to iOS and Android.
```
Comment (erdemyerebasmaz) [inline src/notifications/custom_notifications.md:3]:
```suggestion
You can further customize the Notification Plugin by handling your own notification types sent to the application via push notifications. Follow the code example below to add these to iOS and Android.
```

File: src/notifications/changing_strings.md [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+
+The Notification Plugin uses a set of identifiers and default strings to display messages when processing push notifications. These default strings can be customised by the application. For example, if you wanted to change the `lnurl_pay_metadata_plain_text`, which sets the LNURL-pay text/plain metadata.
```
Comment (erdemyerebasmaz) [inline src/notifications/changing_strings.md:3]: It looks like we're showing how to modify `lnurl_pay_metadata_plain_text` on the following examples. Perhaps it's better to word this as:
```suggestion
The Notification Plugin uses a set of identifiers and default strings to display messages when processing push notifications. These default strings can be customised by the application. For example, to change the `lnurl_pay_metadata_plain_text`, which sets the LNURL-pay text/plain metadata, you can modify it as follows:
```

File: src/notifications/changing_strings.md [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+
+## Android
+
+In the `string.xml` file of the application's `res/values` directory, add the key/value:
```
Comment (erdemyerebasmaz) [inline src/notifications/changing_strings.md:7]:
```suggestion
In the `string.xml` file of the application's `res/values` directory, add the key/value pair:
```

File: src/notifications/changing_strings.md [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="lnurl_pay_metadata_plain_text">Pay to Custom App</string>
+</resources>
+```
+You can find the current identifiers and default strings [here](https://github.com/breez/breez-sdk-liquid/blob/main/lib/bindings/langs/android/lib/src/main/kotlin/breez_sdk_liquid_notification/Constants.kt)
+
+## iOS
+
+In the `Info.plist` file of the `NotificationService` target, add the key/value:
```
Comment (erdemyerebasmaz) [inline src/notifications/changing_strings.md:19]:
```suggestion
In the `Info.plist` file of the `NotificationService` target, add the key/value pair:
```

File: src/notifications/custom_messages.md [hunk 21]
```diff
@@ -0,0 +1,65 @@
...
+The Notification Plugin by default handles a specific format of push notification message data sent by the NDS:
+```json
+{
+    "notification_type": "swap_updated",
+    "notification_payload": "{ \"id\": \"\", \"status\": \"\" }",
+    "app_data": ""
+}
+```
+
+If you want to customise the structure or field names of the push notification message data handled by the Notification Plugin, you can follow the code example below for adding these to iOS and Android. In these examples we replace the `notification_type` and `notification_payload` fields with `custom_type` and `custom_payload`.
```
Comment (erdemyerebasmaz) [inline src/notifications/custom_messages.md:12]: This explanation is a bit wordy? It's better to keep it brief and let the examples speak for themselves imo.
```suggestion
To customize the push notification message data handled by the Notification Plugin, follow the code examples below for iOS and Android. In these examples, the `notification_type` and `notification_payload` fields are replaced with `custom_type` and `custom_payload`.
```

**Review by erdemyerebasmaz — COMMENTED** [review]
Did a first pass.

File: src/notifications/getting_started.md [hunk 22]
```diff
@@ -0,0 +1,77 @@
...
+```
+The structure and fields of this data can be changed by [customising the push messages](custom_messages.md) handling in the Notification Plugin to reflect how your NDS sends this data over push notifications.
+
+## Use cases
+
+The Notification Plugin handles several use cases by default to automatically process push notifications sent via the NDS from when an SDK service calls the registered webhook. If your use case isn't covered by the Notification Plugin, you can extend the plugin to [handle custom notifications](custom_notifications.md).
+
+#### Updating a swap
+
+When making a payment via lightning or a Bitcoin onchain address, the SDK uses a swap service to convert between the Liquid sidechain and Bitcoin onchain or lightning. A swap service is used to monitor changes to the swap status and send push notifications to wake the device to progress the swap. The Notification Plugin when receiving this notification from the NDS will start the SDK and let the SDK complete the swap or progress it to the next state.
```
Comment (erdemyerebasmaz) [inline src/notifications/getting_started.md:32]:
```suggestion
When making a payment via lightning or a Bitcoin onchain address, the SDK uses a swap service to convert between the Liquid sidechain and Bitcoin onchain or Lightning. A swap service is used to monitor changes to the swap status and send push notifications to wake the device to progress the swap. When the Notification Plugin receives this notification from the NDS, it will start the SDK and let the SDK complete the swap or progress it to the next state. 
```

File: src/notifications/getting_started.md [hunk 23]
```diff
@@ -0,0 +1,77 @@
...
+    "data": {  
+        "id": "",    // The hashed id of the swap
+        "status": "" // The latest swap status
+    }
+}
+```
+
+#### Handling LNURL pay requests
+
+Having the ability to process push notifications when the application is in the background or closed also opens up the ability to handle payment requests from a static LNURL or lightning address. To do this the application also needs to register a webhook with an [LNURL-pay service](/guide/lnurl_pay_service.md), then when the LNURL service receives a request on the static LNURL address, it will forward it via the NDS to the application. The Notification Plugin handles the two-step flow for fulfilling these requests.
```
Comment (erdemyerebasmaz) [inline src/notifications/getting_started.md:47]:
```suggestion
Having the ability to process push notifications when the application is in the background or closed also opens up the ability to handle payment requests from a static LNURL or Lightning address. To do this, the application also needs to register a webhook with an [LNURL-pay service](/guide/lnurl_pay_service.md), then when the LNURL service receives a request on the static LNURL address, it will forward it via the NDS to the application. The Notification Plugin handles the two-step flow for fulfilling these requests.
```

File: src/notifications/getting_started.md [hunk 24]
```diff
@@ -0,0 +1,77 @@
...
+
+Firstly the LNURL service receives a request for LNURL-pay information to get the min/max amount that can be received. The LNURL service calls the registered webhook and when receiving this notification, the Notification Plugin will connect to the Breez SDK and send a response back to the LNURL service based on the swap service limits.
```
Comment (erdemyerebasmaz) [inline src/notifications/getting_started.md:49]:
```suggestion
Firstly the LNURL service receives a request for LNURL-pay information to get the min/max amount that can be received. The LNURL service calls the registered webhook, and upon receiving this notification, the Notification Plugin will connect to the Breez SDK and send a response back to the LNURL service based on the swap service limits. 
```

File: src/notifications/getting_started.md [hunk 25]
```diff
@@ -0,0 +1,77 @@
...
+```json
+{
+    "template": "lnurlpay_info",
+    "data": {  
+        "callback_url": "", // The URL of the LNURL service
+        "reply_url": ""     // The URL to reply to this request
+    }
+}
+```
+Secondly the LNURL service receives a request for an invoice based on the selected amount to pay. The LNURL service calls the registered webhook and when receiving this notification, the Notification Plugin will start the SDK and call receive payment for the requested amount. The resulting invoice is then returned to the LNURL service.
```
Comment (erdemyerebasmaz) [inline src/notifications/getting_started.md:61]:
```suggestion
Secondly the LNURL service receives a request for an invoice based on the selected payment amount. The LNURL service calls the registered webhook, and upon receiving this notification, the Notification Plugin will start the SDK and call receive payment for the requested amount. The resulting invoice is then returned to the LNURL service. 
```

File: src/notifications/ios_plugin.md [hunk 26]
```diff
@@ -0,0 +1,70 @@
...
+
+Once added to the Podfile, run `pod install` to install the dependencies.
+
+More installation methods, including with the Swift Package Manager, can be found in the [breez-sdk-liquid-swift](https://github.com/breez/breez-sdk-liquid-swift/blob/main/README.md) repository.
+
+## Integrate the Notification Plugin
+
+You're ready to add some Swift code to implement the Notification Plugin in your NotificationService target. In Xcode, in the `NotificationService` folder, open the Swift file named `NotificationService.swift`.
+
+This Swift file should implement the Notification Plugin's `SDKNotificationService` class. The `SDKNotificationService` class handles the incoming notification content and processes the event. To properly implement this class the NotificationService needs to override at least the `getConnectRequest` function. The `getConnectRequest` function is called by the `SDKNotificationService` to get a `ConnectRequest` which contains the data necessary to connect to the SDK. This data includes the `Config` with it's `workingDir` and the mnemonic.
```
Comment (erdemyerebasmaz) [inline src/notifications/ios_plugin.md:20]:
```suggestion
This Swift file should implement the Notification Plugin's `SDKNotificationService` class. The `SDKNotificationService` class handles the incoming notification content and processes the event. To properly implement this class, the NotificationService needs to override at least the `getConnectRequest` function. The `getConnectRequest` function is called by the `SDKNotificationService` to get a `ConnectRequest` which contains the data necessary to connect to the SDK. This data includes the `Config` with its `workingDir` and the mnemonic.
```

File: src/notifications/ios_plugin.md [hunk 27]
```diff
@@ -0,0 +1,70 @@
...
+
+<div class="warning">
+<h4>Developer note</h4>
+When using the Notification Plugin in iOS, it is important to note that the <code>Config</code> <code>workingDir</code> needs to be set to the app group's shared directory in both the <code>NotificationService</code> target and in the main application target, whether that is a Swift, Flutter or React Native based application.
```
Comment (erdemyerebasmaz) [inline src/notifications/ios_plugin.md:24]:
```suggestion
When using the Notification Plugin on iOS, it is important to note that the <code>Config</code> <code>workingDir</code> needs to be set to the app group's shared directory in both the <code>NotificationService</code> target and in the main application target, whether that is a Swift, Flutter or React Native based application.
```

File: src/notifications/ios_service.md [hunk 28]
```diff
@@ -0,0 +1,57 @@
+# Setting up the Notification Service Extension
+
+## Add a Notfication Service Extension target
```
Comment (erdemyerebasmaz) [inline src/notifications/ios_service.md:3]: typo
```suggestion
## Add a Notification Service Extension target
```

File: src/notifications/ios_service.md [hunk 29]
```diff
@@ -0,0 +1,57 @@
...
+2. Enable the `App Groups` app service
+3. Click `Edit` and select the created app group
+
+### Create Service Extension Identifiers
+Create new release and debug Identifiers for the new service extension.
+1. Click `+` to add an Identifier
+2. Select `App IDs` from the list
+3. Give the Identifier a description and identifier the same as the service extension, like `com.example.application.NotificationService`
+4. Enable the `App Groups` app service
+5. Save the Identifier, then edit again it to add the created app group as above
```
Comment (erdemyerebasmaz) [inline src/notifications/ios_service.md:31]: typo
```suggestion
5. Save the Identifier, then edit it again to add the created app group as above
```

File: src/notifications/ios_service.md [hunk 30]
```diff
@@ -0,0 +1,57 @@
...
+3. Give the name `NotificationService`
+
+This should create a new target with a bundle identifier like `com.example.application.NotificationService` and a folder in your application called `NotificationService`.
+
+## Identifiers
+
+To share data between the main application and service extension targets, you need to add a common `app group` to the main application identifier and new service extension identifier.  Go the [Apple Developer](https://developer.apple.com/account/resources/identifiers/list/applicationGroup) portal, in the Identifiers section select the App Groups dropdown to the right, then:
+1. Click `+` to add an App Group
+2. Select `App Groups` from the list
+3. Give the App Group a description and identifier, like `group.com.example.application`
```
Comment (erdemyerebasmaz) [inline src/notifications/ios_service.md:17]: I think it's better to use a more formal language such as `such as` instead of "like <-example->" on this page :)

File: src/notifications/logging.md [hunk 31]
```diff
@@ -0,0 +1,203 @@
+# Adding logging
+
+You can override the default logger used by the Notification Plugin to use your own logging implementation or dependency. Lets look at an example of a file logger.
```
Comment (erdemyerebasmaz) [inline src/notifications/logging.md:3]:
```suggestion
You can override the default logger used by the Notification Plugin to use your own logging implementation or dependency. Let's look at an example of a file logger.
```

File: src/notifications/logging.md [hunk 32]
```diff
@@ -0,0 +1,203 @@
...
+
+<custom-tabs category="lang">
+<div slot="title">Swift</div>
+<section>
+In iOS lets use the <code>XCGLogger</code> to handle logging to a separate log file in the app group's shared directory.
```
Comment (erdemyerebasmaz) [inline src/notifications/logging.md:8]:
```suggestion
On iOS, let's use the <code>XCGLogger</code> to handle logging to a separate log file in the app group's shared directory.
```

File: src/notifications/logging.md [hunk 33]
```diff
@@ -0,0 +1,203 @@
...
+            self.logger.log(tag: TAG, line:"Failed to set log stream: \(e)", level: "ERROR")
+        }
+    }
+}
+```
+
+</section>
+<div slot="title">Kotlin</div>
+<section>
+In Android lets use the <code>tinylog</code> to handle logging to a file. First create a utility class to configure tinylog.
```
Comment (erdemyerebasmaz) [inline src/notifications/logging.md:46]:
```suggestion
On Android, let's use the <code>tinylog</code> to handle logging to a file. First create a utility class to configure tinylog.
```

File: src/notifications/logging.md [hunk 34]
```diff
@@ -0,0 +1,203 @@
...
+        // Initialise the logger
+        configureLogger(applicationContext)
+        Logger.tag(TAG).debug { "FCM message received!" }
+
+        // Go on to call `startServiceIfNeeded`
+    }
+}
+```
+
+When the foreground service is created, initialise the SDK logger and subscribe to log entries
```
Comment (erdemyerebasmaz) [inline src/notifications/logging.md:105]: Missing punctuation
```suggestion
When the foreground service is created, initialise the SDK logger and subscribe to log entries.
```

File: src/notifications/setup_nds.md [hunk 35]
```diff
@@ -0,0 +1,27 @@
...
+```
+
+The need to run your own NDS is because it's configured to send push notifications to your application users and therefore should be configured with the required keys and certificates. You can use our [reference NDS implementation](https://github.com/breez/notify) as a starting point or as is. Our implementation of the NDS expects URLs in the following format:
+```
+https://your-nds-service.com/notify?platform=<ios|android>&token=[PUSH_TOKEN]
+```
+  
+
+
+This is the same format used when [using webhooks](using_webhooks.md) in the SDK, replacing the `PUSH_TOKEN` with the mobile push token. Once the NDS has received such request it will send a push notification to the corresponding device.
```
Comment (erdemyerebasmaz) [inline src/notifications/setup_nds.md:22]:
```suggestion
This is the same format used when [using webhooks](using_webhooks.md) in the SDK, replacing the `PUSH_TOKEN` with the mobile push token. Once the NDS receives such a request it will send a push notification to the corresponding device.
```

File: src/notifications/setup_nds.md [hunk 36]
```diff
@@ -0,0 +1,27 @@
...
+
+## Mobile push token
+Ensure that your mobile application is set up to receive push notifications and can generate a push token. This token uniquely identifies the device for push notifications.
```
Comment (erdemyerebasmaz) [inline src/notifications/setup_nds.md:25]:
```suggestion
Ensure that your mobile application is set up to receive push notifications and can generate a push token, which uniquely identifies the device for push notifications.
```

File: src/notifications/setup_plugin.md [hunk 37]
```diff
@@ -0,0 +1,7 @@
+# Integrating the plugin
+
+The Notification Plugin is designed to be used when the application isn't running. For this to be achieved reliably and to ensure that the notification is processed, specific implementations are required for both Android and iOS platforms to handle the incoming push notifications.
```
Comment (erdemyerebasmaz) [inline src/notifications/setup_plugin.md:3]:
```suggestion
The Notification Plugin is designed to be used when the application isn't running. To achieve this reliably and ensure that the notification is processed, specific implementations are required for both Android and iOS platforms to handle the incoming push notifications.
```

File: src/notifications/using_webhooks.md [hunk 38]
```diff
@@ -0,0 +1,99 @@
+# Using Webhooks
+
+## Registering a Webhook
+
+Once your vendor [NDS is set up](setup_nds.md) and can accept POST requests from the SDK services, you can within your main application register the webhook URL with the SDK by calling the register webhook API as follows:
```
Comment (erdemyerebasmaz) [inline src/notifications/using_webhooks.md:5]: I think the snippets make it clear that the register webhook API is part of SDK.
```suggestion
Once your vendor [NDS is set up](setup_nds.md) and can accept POST requests from the SDK services, you can register the webhook URL within your main application by calling the register webhook API as follows
```

File: src/notifications/using_webhooks.md [hunk 39]
```diff
@@ -0,0 +1,99 @@
...
+<section>
+
+```dart,ignore
+{{#include ../../snippets/dart_snippets/lib/webhook.dart:unregister-webook}}
+```
+</section>
+</custom-tabs>
+<div class="warning">
+<h4>Developer note</h4>
+Any created payments that use a swap service will continue to use the same registered webhook URL until the swap is complete.
```
Comment (erdemyerebasmaz) [inline src/notifications/using_webhooks.md:98]:
```suggestion
Any payments that use a swap service will use the same registered webhook URL until the swap is complete.
```

**Review by erdemyerebasmaz — CHANGES_REQUESTED** [review]
Did a second pass

File: src/SUMMARY.md (hunk 3 above)
Comment (dangeross) [reply src/SUMMARY.md:48]: I understand, but I'm not going to change the titles as they are approved for Greenlight SDK and Roy was very particular about it

File: src/guide/lnurl_pay_service.md [hunk 40]
```diff
@@ -0,0 +1,91 @@
...
+ "template": "lnurlpay_info",
+ "data": {  
+  "reply_url": https://app.domain.com/respond/<request_id>
+  "callback_url": https://app.domain.com/lnurlpay/invoice
+  }
+}
+```
+
+The <b>reply_url</b> is used by the app to respond to the lnurlpay request.
+The <b>callback_url</b> s the LNURL-Pay callback URL, used by the payer to fetch the invoice.
```
Comment (dangeross) [inline src/guide/lnurl_pay_service.md:60]:
```suggestion
The ```reply_url``` is used by the app to respond to the LNURL-Pay request.
The ```callback_url``` is the LNURL-Pay callback URL, used by the payer to fetch the invoice.
```

File: src/notifications/getting_started.md [hunk 41]
```diff
@@ -0,0 +1,77 @@
...
+
+## How it works
+
+The process involves using a Notification Delivery Service (NDS) acting as an intermediary host by the application developer. The NDS must provide a public facing webhook URL where a POST request can be sent to when a notification needs to be delivered to the application. The NDS then forwards the data sent in the webhook POST request via push notification to the application. When the application then receives the push notification, the SDK Notification Plugin can be used to process the event.
+
+![Breez SDK - Liquid Notifications](../images/BreezSDK_Liquid_Notifications.png)
+
+### Push notification data
+
+When receiving POST request data on the webhook URL, the NDS should get the `token`, `platform` and optional `app_data` from the URL params. If necessary, convert the token to a token that can be sent to the push notification service. Then format the data to be sent to the push notification service, which includes converting any JSON data into a string format.
```
[the remark this answers is not in the document]
Comment (dangeross) [reply src/notifications/getting_started.md:13]: For example, if the application uses a non push service token identifier as the token in the webhook URL, then when receiving the POST request if does an internal lookup to map the `token` to the push service `token`

File: src/notifications/ios_plugin.md [hunk 42]
```diff
@@ -0,0 +1,70 @@
...
+            os_log(.error, "Mnemonic not found")
+            return nil
+        }
+        return ConnectRequest(config: config, mnemonic: mnemonic)
+    }
+}
+```
+
+## Reference implementation
+For a complete reference, see how we implemented it in misty-breez wallet: [NotificationService.swift](https://github.com/breez/misty-breez/blob/main/ios/Breez%20Notification%20Service%20Extension/NotificationService.swift).
```
Comment (dangeross) [inline src/notifications/ios_plugin.md:70]:
```suggestion
For a complete reference, see how we implemented it in misty-breez wallet: [NotificationService.swift](https://github.com/breez/misty-breez/blob/main/ios/NotificationService/NotificationService.swift).
```

File: src/notifications/ios_service.md (hunk 30 above)
Comment (dangeross) [inline src/notifications/ios_service.md:17]:
```suggestion
3. Give the App Group a description and identifier, such as `group.com.example.application`
```

File: src/notifications/getting_started.md (hunk 41 above)
[the remark this answers is not in the document]
Comment (erdemyerebasmaz) [reply src/notifications/getting_started.md:13]: Perhaps we can add this to docs for clarification?
```suggestion
When receiving POST request data on the webhook URL, the NDS should get the `token`, `platform` and optional `app_data` from the URL params. If necessary, convert the token into a token compatible with the push notification service. For example, if the application uses a different token identifier in the webhook URL, it will perform an internal lookup to convert this token into a push notification service token upon receiving the POST request. Then format the data to be sent to the push notification service, which includes converting any JSON data into a string format. 
```

**Review by erdemyerebasmaz — APPROVED** [review]

**Review by erdemyerebasmaz — APPROVED** [review]
LGTM!

Can be merged after upgrading swift package & CI passes.

**Review by erdemyerebasmaz — APPROVED** [review]
